akleventis: update extension

akleventis opened the pull request:
styles: 
--------
- style changes for more compact look
- colors to match google tab groups
- drop-down 3x3 grid with hover effects
- updated logo (brighter, more visible with light/dark mode)
<p align="center">
<img src="https://user-images.githubusercontent.com/64287087/157167653-95555761-3a12-4e8c-b502-2796ace92c87.gif" width="500px" />
</p>
<br/>

--------

<br/>

functionality
--------
- add 3 more groups
- ability to "go back" out of select screen
- ***ungroup tab if no matching text in url*** 
<p align="center">
 <img src="https://user-images.githubusercontent.com/64287087/157167677-2fb297da-9b06-4b90-ac87-66707c7a68b5.gif" width="500px" />
</p>

akleventis committed:
styles/functionality upgrade

zacharytyhacz commented:
Hello @akleventis - I am not familiar with testing/developing extensions but can you also test on Brave ( chromium, very similar to Chrome ) ? The tabs are a little different so your new styles might look a little different / broken on Brave

How latest version from chrome store looks for me on Brave:

![image](https://user-images.githubusercontent.com/38140593/157247707-b2b6d5d4-df2b-467e-a150-1058a6adebfe.png)

akleventis commented:
sounds good! I'll test it out on Brave and get back to you soon ✓ @zacharytyhacz

furofo commented:
I will take a look at later too I'm on vacation this week but what but will
have time to look at when I get back I'm a week looks goof though!

On Tue, Mar 8, 2022, 9:32 AM Alex ***@***.***> wrote:

> sounds good! I'll test it out on Brave and get back to you soon ✓
> @zacharytyhacz <https://github.com/zacharytyhacz>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/furofo/TabGroupExtension/pull/2#issuecomment-1061840294>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACLXCO3QP44RAT6GHLUFZVDU65QIPANCNFSM5QFIDE3A>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

akleventis committed:
fix delete

akleventis commented:
Brave Browser ✅

https://user-images.githubusercontent.com/64287087/157290798-fff8aaca-ec4f-40a4-a635-ae8a58c76200.mov

zacharytyhacz commented on js/globalValAndFunction.js:
```diff
@@ -218,6 +218,7 @@ let deleteButtonLogic = (isCheckedArray, tabGroupsArray, dropDownBox, isback) =>
         // end here if not just return and do nothing
         if (i === isCheckedArray.length - 1) {
           if (isCheckedArray[i].checked) {
+            console.log("IN HEER !")
```
lol

akleventis replied:
I'll fix that soon haha @zacharytyhacz

akleventis committed:
remove super ultra professional log

akleventis committed:
start refactor of readme.. will update more if changes are merged (:

zacharytyhacz commented on js/popup.js:
```diff
@@ -17,6 +17,12 @@ deleteButton.addEventListener('click', async () => {
+gobackButton.addEventListener('click', async () => {
+  const isCheckedArray = document.querySelectorAll('.container input');
+  const dropDownBox = document.querySelectorAll('.dropdown');
+  deleteButtonLogic(isCheckedArray, tabGroupsArray, dropDownBox, true)
```
I've tested the extension on my browser and when I select multiple rows + enter my data, but with one row empty with no data when I hit Go Back to cancel, the inputs have weird behavior:

<img width="380" alt="image" src="https://user-images.githubusercontent.com/38140593/158492787-cff9074a-4a46-44ca-a0aa-d7d44889c11e.png">

Then, when I attempt to edit these rows the toggling of the input's state does not allow me to input anything:

( I am seemingly in 'edit mode' here )
<img width="377" alt="image" src="https://user-images.githubusercontent.com/38140593/158492867-7a03f8e0-1da3-4114-9d47-820cdda29dba.png">


I recommend adding specific functionality for the `goBackButton` click event instead of using the `deleteButtonLogic` function which seems to do something totally different than simply 'going back'.

akleventis replied:
aw bummer! Thought I could cheat a bit with that go back functionality. Thanks for the update! I'll go ahead and separate the functions  @zacharytyhacz

akleventis committed:
add 'go-back' function, helper functions, add 2 more groups

akleventis committed:
additional component

akleventis committed:
Update readme.md

akleventis commented:
*phew*
- Majority of this commit was introducing a `go back` function
- Uses global variable `tabGroupArray` to set values back to original state
- I tested many scenarios and couldn't find a way to break it. However a new set of eyes on testing may introduce new bugs
- Added helper functions for code cleanup alongside comment refactors
- Added 2 more groups (max number chromium supports)
- Updated readme with new pictures and formatting

https://user-images.githubusercontent.com/64287087/159757243-d5b31552-a4c4-4957-84a8-5e0fc324f614.mov

akleventis commented:
would you guys mind checking this out ^ ? Would be cool to release the updated extension 🙏🏼 @zacharytyhacz @furofo

thanks!

furofo commented:
Hey sorry for the late reply. I will definitely check this out and release
it but I am going to need some time. I have a lot going on currently and
won't be able to really dig into this again for about a month.

On Thu, Apr 7, 2022 at 9:55 AM Alex Leventis ***@***.***>
wrote:

> would you guys mind checking this out ^ ? Would be cool to release the
> updated extension 🙏🏼 @zacharytyhacz <https://github.com/zacharytyhacz>
> @furofo <https://github.com/furofo>
>
> thanks!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/furofo/TabGroupExtension/pull/2#issuecomment-1091770701>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACLXCO6VOWFY5V664TM7C53VD3SM3ANCNFSM5QFIDE3A>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

akleventis commented:
hey no worries at all, thanks for replying! Take all the time you need @furofo

akleventis committed:
replace repetitive assigning click functions with loop
